rfcs/QueryLevelNullability.md

```diff
@@ -1,100 +1,158 @@
-that is used to express when developers can be certain that a value can or can never be `null` 
-(or the language equivalent). Nullability language constructs (eg `?` in Swift/Kotlin)
-have become popular due to their ability to solve ergonomic problems in languages
-such as those surrounding `NullPointerException` in Java.
+- **Nullability**. A feature of many programming languages (eg [Swift](https://developer.apple.com/documentation/swift/optional),
+  [Kotlin](https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-null-types), [SQL](https://www.w3schools.com/sql/sql_notnull.asp))
+  that is used to indicate whether or not a value can be `null`.
 
-Codegen: Short for Code Generation, tools that exist for working with GraphQL which accept a schema and a set of documents as
-input, and output code in a language of choice that represents the data returned by those 
-operations. GraphQL codegen tools exist for many platforms. As an example, [here's some info about Apollo's offerings](https://github.com/apollographql/apollo-tooling#code-generation) and [GraphQL Codegen by The Guild](https://www.graphql-code-generator.com/).
+  Nullability language constructs (e.g. `?` in Swift/Kotlin) have become popular due to their ability to solve ergonomic
+  problems in programming languages, such as those surrounding `NullPointerException` in Java.
 
-## 📜 Problem Statement
+- **Codegen**. Short for "code generation", in this proposal refers to tools that generate code to facilitate using
+GraphQL on the client. GraphQL codegen tooling exists for many platforms:
+  - [Apollo](https://github.com/apollographql/apollo-tooling#code-generation) has a code generator for Android (Kotlin)
+    and iOS (Swift) clients
+  - [The Guild](https://www.graphql-code-generator.com/) has a TypeScript code generator for web clients
 
-According to [best practice](https://graphql.org/learn/best-practices/#nullability) GraphQL is
-a "nullable by default" language meaning that all properties are allowed to be `null`.
-This is in contrast to the two modern languages used on mobile clients, Swift and Kotlin,
-which are both non-null by default. In Swift and Kotlin, unless developers otherwise
-specify it, properties cannot be `null`.
+  GraphQL codegen tools typically accept a schema and a set of documents as input, and output code in a language of
+  choice that represents the data returned by those operations.
 
-This mismatch creates some dissonance for developers who are currently forced to deal with
-problems that commonly surround nullablility. This makes large numbers of null fields tedious and
-time-consuming to work with.
+  For example, the Apollo iOS codegen tool generates Swift types to represent each query document, as well as model types
+  representing the data returned from those queries. Notably, a nullable field in the schema becomes an `Optional`
+  property on the generated Swift model type, represented by `?` following the type name.
```

Comment:
I decided to use our "canonical" example in the definition section, since we refer to it throughout, but we could move it to an Appendix section and refer to it that way instead (to declutter the definition section)